controller/lib/core/blower_fsm.cpp

```diff
@@ -90,92 +86,87 @@ static Duration ExpireDuration(const VentParams &params) {
 PressureControlFsm::PressureControlFsm(Time now, const VentParams &params)
     : inspire_pressure_(cmH2O(static_cast<float>(params.pip_cm_h2o))),
       expire_pressure_(cmH2O(static_cast<float>(params.peep_cm_h2o))),
-      setpoint_(expire_pressure_), start_time_(now),
-      inspire_end_(start_time_ + InspireDuration(params)),
+      start_time_(now), inspire_end_(start_time_ + InspireDuration(params)),
       expire_end_(inspire_end_ + ExpireDuration(params)) {}
 
-void PressureControlFsm::Update(Time now, const BlowerFsmInputs &inputs) {
+BlowerSystemState
+PressureControlFsm::DesiredState(Time now, const BlowerFsmInputs &inputs) {
+  bool is_end_of_breath = false;
+  if (now >= expire_end_) {
+    is_end_of_breath = true;
+  }
   if (now >= inspire_end_) {
-    inspire_finished_ = true;
+    return {
+        .pressure_setpoint = expire_pressure_,
+        .flow_direction = FlowDirection::EXPIRATORY,
+        .pip = inspire_pressure_,
+        .peep = expire_pressure_,
+        .is_end_of_breath = is_end_of_breath,
+    };
   } else {
     // Go from expire_pressure_ to inspire_pressure_ over a duration of
     // RISE_TIME.  Then for the rest of the inspire time, hold at
     // inspire_pressure_.
     static_assert(RISE_TIME > milliseconds(0));
     float rise_frac = std::min(1.f, (now - start_time_) / RISE_TIME);
-    setpoint_ =
+    Pressure setpoint =
```

Comment:
nit, get rid of this variable and move the computation into the return statement?